backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/CodeVersionHandler.java

```diff
@@ -0,0 +1,118 @@
+package ai.verta.modeldb.experimentRun.subtypes;
+
+import ai.verta.modeldb.CodeVersion;
+import ai.verta.modeldb.LogExperimentRunCodeVersion;
+import ai.verta.modeldb.ModelDBConstants;
+import ai.verta.modeldb.common.futures.FutureJdbi;
+import ai.verta.modeldb.common.futures.InternalFuture;
+import ai.verta.modeldb.entities.CodeVersionEntity;
+import ai.verta.modeldb.exceptions.AlreadyExistsException;
+import ai.verta.modeldb.utils.ModelDBHibernateUtil;
+import ai.verta.modeldb.utils.RdbmsUtils;
+import java.util.Optional;
+import java.util.concurrent.Executor;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.hibernate.LockMode;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+public class CodeVersionHandler {
```

Comment:
I have already implemented this class.

Reply:
Yes, this is extracted from your PR

Reply:
Just point your branch to base off mine then. Let's not copy code between branches as it makes things much harder to manage.

Reply:
Updated branch based with your code